### PR TITLE
[PRIVILEGES] :sparkles: Add user Privileges

### LINKED ISSUE #3: Add user Privileges
- Allow visibility according to user privileges
- Add the user privileges to the Token

### PR DESCRIPTION
**Pull request by DjihadBengati**
Fixes #3

### CODE REVIEW

File: src/main/java/org/openemp/api/user/model/Role.java [hunk 1]
```diff
@@ -1,16 +1,13 @@
...
-import lombok.Setter;
-
-import javax.persistence.CascadeType;
-import javax.persistence.Entity;
-import javax.persistence.FetchType;
-import javax.persistence.OneToMany;
 import java.util.HashSet;
 import java.util.Set;
 
+import javax.persistence.*;
```
Comment (chinzou) [inline src/main/java/org/openemp/api/user/model/Role.java]: @DjihadBengati, Can you please avoid using wild cards ("*") in imports, for readability reasons.

File: src/main/java/org/openemp/api/user/model/Role.java [hunk 2]
```diff
@@ -19,10 +16,20 @@
 @Setter
 public class Role extends BaseEntity {
 
-    private String name;
+	private static final long serialVersionUID = -6132338183111628177L;
 
-    @OneToMany(cascade = CascadeType.ALL,
```
Comment (chinzou) [inline src/main/java/org/openemp/api/user/model/Role.java]: Is there a reason why this is not aligned?

File: src/test/java/org/openemp/api/user/OpenempUserApiApplicationTests.java [hunk 3]
```diff
@@ -1,19 +1,102 @@
...
+	public void setup() {
+		UserController userController = new UserController(userService, authenticationManager, jwtTokenUtil,
+				userDetailsService);
+		this.mockMvc = MockMvcBuilders.standaloneSetup(userController).build();
+	}
+
 	@Test
-    void contextLoads() {
-    }
+	public void testAuthorize() throws Exception {
```
Comment (chinzou) [inline src/test/java/org/openemp/api/user/OpenempUserApiApplicationTests.java]: Can we have these tests in a separate file, something like **OpenEMPSecurityTests.java**

**Review by chinzou — CHANGES_REQUESTED** [review]

File: src/main/java/org/openemp/api/user/model/Role.java (hunk 2 above)
Comment (DjihadBengati) [reply src/main/java/org/openemp/api/user/model/Role.java]: What do you mean by aligned ?

File: src/test/java/org/openemp/api/user/OpenempUserApiApplicationTests.java (hunk 3 above)
Comment (DjihadBengati) [reply src/test/java/org/openemp/api/user/OpenempUserApiApplicationTests.java]: The testAutorize in the OpenEMPSecurityTests.java ?

Comment (chinzou) [reply src/test/java/org/openemp/api/user/OpenempUserApiApplicationTests.java]: yes, it make sense. isn't?

File: src/main/java/org/openemp/api/user/model/Role.java (hunk 2 above)
Comment (chinzou) [reply src/main/java/org/openemp/api/user/model/Role.java]: I mean, they just look like not at the same level lol, something like this
```           
         private .......
       @OneTo...
```

File: src/test/java/org/openemp/api/user/OpenempUserApiApplicationTests.java (hunk 3 above)
Comment (DjihadBengati) [reply src/test/java/org/openemp/api/user/OpenempUserApiApplicationTests.java]: Yes why not ! I will send a commit doing that :)

**Review by chinzou — APPROVED** [review]

**Comment by chinzou** [comment]
Thanks @DjihadBengati

**Pull request merged**